docs/components/index.md

```diff
@@ -0,0 +1,310 @@
+}
+```
+
+## Component disposal with IDisposable
+
+If a component implements [IDisposable](/dotnet/api/system.idisposable), the router disposes the component when the user navigates away from the component. A component can implement `IDisposable` in a C# base class that implements `BlazorComponent` (or `IComponent` directly). The following component uses `@implements IDisposable` and the `Dispose` method:
+
+```csharp
+@using System
+@implements IDisposable
+
+...
+
+@functions {
+    public void Dispose()
+    {
+        ...
+    }
+}
+```
+
+## Render child content
+
+Blazor components can render child content with an element that matches the child component's name. The following Index page renders the content of the `HeadingComponent` (*HeadingComponent.cshtml*):
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/Index.cshtml?start=1&end=11)]
+
+A child component can receive property assignments from their calling parent. In the following example, the `ParentComponent` sets the value of the `Title` property in the `ChildComponent`:
+
+*ParentComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ParentComponent.cshtml?start=1&end=7)]
+
+*ChildComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ChildComponent.cshtml)]
+
+The body of the Bootstrap-styled panel is provided by `ChildContent`. `ChildContent` is bound to the `<ChildComponent>` element's content in the parent component by convention.
+
+## Routing
+
+Routing in Blazor is achieved by providing a route template to each accessible component in the app. Route templates are converted into [RouteAttribute](/dotnet/api/microsoft.aspnetcore.mvc.routeattribute) implementations. Multiple route templates can be applied to a component. The following component responds to requests for `/BlazorRoute` and `/DifferentBlazorRoute`:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRoute.cshtml?start=1&end=4)]
+
+## Route parameters
+
+Blazor components can receive and use route parameters from the route template provided in the `@page` directive:
+
+*RouteParameter.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/RouteParameter.cshtml?start=1&end=8)]
+
+Not all route parameter constraints are supported at this time. Optional parameters aren't currently supported, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
+
+For more information on attribute routing, see [Routing](xref:fundamentals/routing).
+
+## JavaScript/TypeScript interop
+
+To call JavaScript libraries or custom JavaScript/TypeScript code from .NET, the current approach is to register a named function in a JavaScript/TypeScript file:
+
+```javascript
+Blazor.registerFunction('doPrompt', message => {
+  return prompt(message);
+});
+```
+
```

Comment:
@guardrex  may be consider mentioning that the JS file or <script> tag with registerFunction invocations should be placed in index.html after the blazor boot script tag and not from the components\Razor themselves. Else you risk running into issues like https://github.com/aspnet/Blazor/issues/451

Reply:
yes, ty @muqeet-khan I'll do that. :+1: I'll get in there on the next commit.